libs/barista-components/drawer-table/src/drawer-table.ts

```diff
@@ -0,0 +1,188 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AfterContentInit,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChild,
+  Directive,
+  Input,
+  TemplateRef,
+  ViewChild,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  DtHeaderRowDef,
+  DtRowDef,
+  DtTable,
+} from '@dynatrace/barista-components/table';
+
+import { DtDrawer } from '@dynatrace/barista-components/drawer';
+import { isEmpty } from '@dynatrace/barista-components/core';
+
+@Component({
+  selector: 'dt-drawer-table',
+  templateUrl: './drawer-table.html',
+  styleUrls: ['./drawer-table.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+})
+export class DtDrawerTable<T> implements AfterViewInit, AfterContentInit {
+  /** Columns that stay visible when row is expanded  */
+  @Input() openColumns: string[];
+
+  /** Reference to the drawer element  */
+  @ViewChild(DtDrawer, { static: true }) private drawer: DtDrawer;
+
+  /** Reference to the drawer container  */
+  @ViewChild('container', { read: ViewContainerRef, static: false })
+  private container: ViewContainerRef;
+
+  /** Template reference for the drawer content. */
+  @ContentChild(TemplateRef, { static: false }) drawerContentRef: TemplateRef<{
+    $implicit: T;
+  }>;
+
+  /** Reference to the table that is used to make the rows interactive */
+  @ContentChild(DtTable, { static: false }) table: DtTable<T>;
+
+  /** Reference to the header row definition which is used to set the columns  */
+  @ContentChild(DtHeaderRowDef, { static: false }) rowHeaderDef: DtHeaderRowDef;
+
+  /** Reference to the row definition which is used to get and set the columns  */
+  @ContentChild(DtRowDef, { static: false }) rowDef: DtRowDef<T>;
+  private initialColumns: Iterable<string>;
```

Comment:
Is there a specific reason why you are using `Iterable` typing here instead of an array? Is the rowDef columns from the cdk also typed as `Iterable`

Reply:
Yes, the reason why I used Iterable was that columns from cdk also types as Iterable. Do you recommend using array instead and converting the columns to array when assigning?

Reply:
No this is fine. I'd rather not convert if the cdk table uses `Iterable`. In the end, the cdk needs to deal with the iteration. Keep it as Iterable please.